integration/vcpus/default_vcpus_test.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This will test the default_vcpus
+# feature is working properly
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+dir_path=$(dirname "$0")
+source "${dir_path}/../../lib/common.bash"
+source "${dir_path}/../../.ci/lib.sh"
+KATA_HYPERVISOR="${KATA_HYPERVISOR:-qemu}"
+
+if [ "${KATA_HYPERVISOR}" == "firecracker" ]; then
+	info "Skip: This is not working see https://github.com/kata-containers/runtime/issues/2317"
+	exit 0
+fi
+
+function setup() {
+	clean_env
+	check_processes
+	extract_kata_env
+}
+
+function test_docker_with_vcpus() {
+	sudo sed -i 's/default_vcpus = 1/default_vcpus = 4/g' ${RUNTIME_CONFIG_PATH}
+	docker run -i --runtime=${RUNTIME} busybox nproc | grep "4"
+	sudo sed -i 's/default_vcpus = 4/default_vcpus = 1/g' ${RUNTIME_CONFIG_PATH}
```

Comment:
OOI, if this fails, we don't revert the change. But, I guess at that point the CI fails anyway, and we tear down the CI node... so probably OK.
I guess you could move the cleanup into the `teardown()` function, and that would catch that case?

Reply:
changes applied